Src/FluentAssertions/Common/MemberPathSegmentEqualityComparer.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System.Collections.Generic;
+using System.Text.RegularExpressions;
+
+namespace FluentAssertions.Common
+{
+    /// <summary>
+    /// Compares two segments of a <see cref="MemberPath"/>.
+    /// Sets the <see cref="AnyIndexQualifier"/> equal with any numeric index qualifier.
+    /// All other comparisons are default string equality.
+    /// </summary>
+    internal class MemberPathSegmentEqualityComparer : IEqualityComparer<string>
+    {
+        private const string AnyIndexQualifier = "*";
+        private static readonly Regex IndexQualifierRegex = new(@"^\d+$");
+
+        /// <summary>
+        /// Compares two segments of a <see cref="MemberPath"/>.
+        /// </summary>
+        /// <param name="x">Left part of the comparison.</param>
+        /// <param name="y">Right part of the comparison.</param>
+        /// <returns>True if segments are equal, false if not.</returns>
+        public bool Equals(string x, string y)
+        {
+            if (x is null || y is null)
+            {
+                return x == y;
+            }
```

Comment:
According to code coverage this branch is never entered.
I believe it is unreachable and can be deleted as `x` and `y` comes from `MemberPath.Segments` which will never contain `null` elements.

Reply:
So if @whymatter removes this one, we can merge.

Reply:
Yes 🤩

Reply:
Please also be aware of the release notes 😉

Reply:
@dennisdoomen done 👍